src/cockpit/channel.py

```diff
@@ -136,6 +138,13 @@ def do_channel_control(self, channel, command, message):
         except ChannelError as exc:
             self.close(**exc.kwargs)
 
+    def do_kill(self, host: Optional[str], group: Optional[str]) -> None:
+        if host is not None:
+            return
+        if self.group is not None and self.group != group:
```

Comment:
Err, but I think we need to cover for `group is None`, and then always kill; see [protocol](https://github.com/cockpit-project/cockpit/blob/main/doc/protocol.md#command-kill). Perhaps that's what you actually intended to write?

Reply:
This condition is false.  I had fixed that up in another commit, which apparently didn't make it in here.  This is meant to read `if group is not None ...` since not specifying a group on a kill message means "kill everything".

Reply:
Ack, this looks a lot better now.